### PR TITLE
docs: update output.clean type declaration

### PR DESCRIPTION
**Pull request by cbbfcd**
```rust
// https://github1s.com/web-infra-dev/rspack/blob/main/crates/rspack_binding_values/src/clean_options.rs#L13
#[napi(object, object_to_js = false)]
#[derive(Debug)]
pub struct JsCleanOptions {
  pub keep: Option<String>,
  // todo:
  // - support RegExp type
  //   if path match the RegExp, keep the file
  // - support function type
  //    if the fn returns true on path str, keep the file
}
```

## Checklist

- [ ] Tests updated (or not required).
- [x] Documentation updated (or not required).

### CODE REVIEW

**Review by chenjiahan — COMMENTED** [review]
Thanks! Could you help by adding an example using `clean.keep` ?

**Comment by cbbfcd** [comment]
> Thanks! Could you help by adding an example using `clean.keep` ?

ok~

File: website/docs/zh/config/output.mdx [hunk 1]
```diff
@@ -161,6 +161,13 @@ module.exports = {
   output: {
     clean: true, // Clean the output directory before emit.
   },
+
+  // or
+  output: {
+    clean: {
+      keep: 'path', // keep the files under this path
```
Comment (chenjiahan) [inline website/docs/zh/config/output.mdx:168]:
```suggestion
      keep: 'ignored/dir', // keep these assets under 'dist/ignored/dir'.
```

File: website/docs/en/config/output.mdx [hunk 2]
```diff
@@ -165,6 +165,13 @@ module.exports = {
   output: {
     clean: true, // Clean the output directory before emit.
   },
+
+  // or
+  output: {
+    clean: {
+      keep: 'path', // keep the files under this path
```
Comment (chenjiahan) [inline website/docs/en/config/output.mdx:172]:
```suggestion
      keep: 'ignored/dir', // keep these assets under 'dist/ignored/dir'.
```

**Review by chenjiahan — APPROVED** [review]
Thanks!

**Pull request merged**